2996_reflection/reflection.md

```diff
@@ -1823,6 +1819,60 @@ Change the grammar for `$operator-or-punctuator$` in paragraph 1 of [lex.operato
 ```
 :::
 
+### [basic.types.general]
+
+Change the first sentence in paragraph 9 of [basic.types.general]{.sref} as follows:
+
+::: bq
+[9]{.pnum} Arithmetic types (6.8.2), enumeration types, pointer types, pointer-to-member types (6.8.4),[ `std::meta::info`,]{.addu} `std::nullptr_t`, and cv-qualified (6.8.5) versions of these types are collectively called scalar types.
+:::
+
+Add a new paragraph at the end of [basic.types.general]{.sref} as follows:
+
+::: bq
+::: addu
+
+[*]{.pnum} A *consteval-only* is one of the following:
+
+  - `std::meta::info`, or
+  - a pointer or reference to a consteval-only type, or
+  - an (possibly multi-dimensional) array of a consteval-only type, or
+  - a type pointer-to-member of class C of type M where C and/or M is a consteval-only type, or
+  - a function type with a consteval-only return type or a consteval-only parameter type, or
+  - a class type with a consteval-only base class type or consteval-only nonstatic data member type.
+
+An object of consteval-only type shall either end its lifetime during the evaluation of a manifestly constant-evaluated expression or conversion ([expr.const]{.sref}), or be a constexpr variable that is not odr-used ([basic.def.odr]{.sref}).
+
+:::
+:::
+
+### [basic.fundamental] Fundamental types
+
+Add a new paragraph before the last paragraph of [basic.fundamental]{.sref} as follows:
+
+::: bq
+::: addu
+
+[*]{.pnum} A value of type `std::meta::info` is called a _reflection_ and represents a language element such as a type, a constant value, a non-static data member, etc.
+`sizeof(std::meta::info)` shall be equal to `sizeof(void*)`.
+[*Note*:
+Reflections are only meaningful during translation.
+The notion of *consteval-only* types (see [basic.types.general]{.sref}) exists to diagnose attempts at using such values outside the translation process.]
+
+:::
+:::
+
+### [basic.lookup.argdep] Argument-dependent name lookup
+
+Add a bullet after the first in paragraph 3 of [basic.lookup.argdep] as follows:
+::: bq
+[3]{.pnum} ... The set of entities is determined in the following way:
+
+- If `T` is a fundamental type, its associated set of entities is empty.
+- [If `T` is `std::meta::info`, its associated set of entities is the singleton containing namespace `std::meta`.]{.addu}
+- If `T` is a class type ...
```

Comment:
Adding sub-bullet numbers:
```suggestion
- [3.1]{.pnum} If `T` is a fundamental type, its associated set of entities is empty.
::: addu
- [3.2]{.pnum} If `T` is `std::meta::info`, its associated set of entities is the singleton containing namespace `std::meta`.
:::
- [3.3]{.pnum} If `T` is a class type ...
```

Reply:
Actually is this right? We say:

> The [associated namespaces](https://eel.is/c++draft/basic.lookup.argdep#def:associated_namespaces) for a call are the innermost enclosing non-inline namespaces for its associated entities as well as every element of the inline namespace set ([[namespace.def]](https://eel.is/c++draft/namespace.def)) of those namespaces.

What's the innermost enclosing non-inline namespace of... `std::meta`, isn't that `std`? Should the associated entities of `std::meta::info` be `{std::meta::info}`? Given:

```cpp
namespace std::meta { using info = decltype(^::); }
```
the innermost enclosing non-inline namespace of `std::meta::info` is still, reasonably, `std::meta` right?

Reply:
I think ADL doesn't capture associated namespaces of an alias, hence the need for a special case.

Reply:
I know there's a need for a special case - but as written this isn't correct. I think the associated entity needs to be `std::meta::info` itself, so that the next rule about associated namespaces picks up `std::meta`.